lib/rules/no-magic-numbers.js

```diff
@@ -79,6 +79,10 @@ module.exports = function(context) {
                 return;
             }
 
+            if (parent.type === "UnaryExpression" && parent.operator === "-") {
```

Comment:
You should move this up above line 69, and any reference to `node.value` should be replaced with a stored value that is negated when we get into this block.